src/ChartCreator/DataSelector/DatasetCreator/DatasetCreator.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import React, { useState, useEffect } from "react";
+import TextField from "@material-ui/core/TextField";
+import InputLabel from "@material-ui/core/InputLabel";
+import Grid from "@material-ui/core/Grid";
+import FormControl from "@material-ui/core/FormControl";
+import Select from "@material-ui/core/Select";
+import IconButton from "@material-ui/core/IconButton";
+import AddIcon from "@material-ui/icons/Add";
+import DeleteIcon from "@material-ui/icons/Delete";
+import { v4 as uuidv4 } from "uuid";
+import ColorPicker from "./ColorPicker";
+import { hexColors, rgbaColors } from "../../utils/colors";
+
+const DatasetCreator = ({ dataset, datasets, setDatasets, buildMenuItems }) => {
+  const [index, setIndex] = useState(0);
+  const [deleteDisabled, setDeleteDisabled] = useState(false);
+  const [addDisabled, setAddDisabled] = useState(false);
+  const { id, ordinate, label, backgroundColor } = dataset;
+
+  useEffect(() => {
+    setIndex(datasets.findIndex((dt) => dt.id === id));
+  }, [id, datasets]);
+
+  useEffect(() => {
+    setDeleteDisabled(datasets.length === 1);
+    setAddDisabled(datasets.length > 1 && index !== datasets.length - 1);
+  }, [datasets, index]);
+
+  const handleDatasetChange = ({ value, key }) => {
+    const newDataSets = [...datasets];
```

Comment:
shallow copy + object mutation?

Reply:
Agree I'll use cloneDeep.
Do you have any idea to avoid object mutation here ? (Can be released in version)